pkg/trace/stats/sublayers.go

```diff
@@ -34,6 +43,181 @@ func (v SublayerValue) GoString() string {
 	return v.String()
 }
 
+// SublayerCalculator holds arrays used to compute sublayer metrics.
+// Re-using arrays reduces the number of allocations
+type SublayerCalculator struct {
+	activeSpans      []int
+	activeSpansIndex []int
+	openSpans        []bool
+	nChildren        []int
+	execDuration     []float64
+	parentIdx        []int
+	timestamps       timestampArray
+	maxSpans         int
+}
+
+// NewSublayerCalculator returns a new SublayerCalculator used to compute sublayer metrics
```

Comment:
```suggestion
// NewSublayerCalculator returns a new SublayerCalculator.
```
The structure's comment contains all the information about what it is. This is just the constructor. It returns a ready-to-use SublayerComputator.